actors/user/src/main/java/org/sunbird/user/actors/UserMergeActor.java

```diff
@@ -0,0 +1,128 @@
+package org.sunbird.user.actors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.HttpUtilResponse;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.*;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.models.user.User;
+import org.sunbird.telemetry.util.TelemetryUtil;
+import org.sunbird.user.service.UserService;
+import org.sunbird.user.service.impl.UserServiceImpl;
+
+import java.io.IOException;
+import java.util.*;
+
+@ActorConfig(
+        tasks = {"mergeUser"},
+        asyncTasks = {}
+)
+public class UserMergeActor extends UserBaseActor {
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private UserService userService = UserServiceImpl.getInstance();
+    @Override
+    public void onReceive(Request userRequest) throws Throwable {
+        updateUserMergeDetails(userRequest);
+    }
+
+    private void updateUserMergeDetails(Request userRequest) throws IOException {
+        ProjectLogger.log(
+                "UserMergeActor:updateUserMergeDetails: starts : " ,
+                LoggerEnum.DEBUG.name());
+        Map<String, Object> targetObject = null;
+        List<Map<String, Object>> correlatedObject = new ArrayList<>();
+        Response response = new Response();
+        Map mergeeESMap = new HashMap<String, Object>();
+        Map mergeeDBMap = new HashMap<String, Object>();
+        Map requestMap = userRequest.getRequest();
+        String mergeeId = (String)requestMap.get(JsonKey.FROM_ACCOUNT_ID);
+        String mergerId = (String)requestMap.get(JsonKey.TO_ACCOUNT_ID);
+        User mergee = userService.getUserById(mergeeId);
+        if(!mergee.getIsDeleted()) {
+            mergeeDBMap.put(JsonKey.STATUS, 0);
+            mergeeDBMap.put(JsonKey.IS_DELETED, true);
+            mergeeDBMap.put(JsonKey.EMAIL,null);
+            mergeeDBMap.put(JsonKey.PHONE,null);
+            mergeeDBMap.put(JsonKey.USERNAME, null);
+            mergeeDBMap.put(JsonKey.PREV_USED_EMAIL, mergee.getEmail());
+            mergeeDBMap.put(JsonKey.PREV_USED_PHONE, mergee.getPhone());
+            mergeeDBMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
+            mergeeDBMap.put(JsonKey.ID,mergee.getId());
+
+            mergeeESMap.put(JsonKey.STATUS, 0);
+            mergeeESMap.put(JsonKey.IS_DELETED, true);
+            mergeeESMap.put(JsonKey.EMAIL,null);
+            mergeeESMap.put(JsonKey.PHONE,null);
+            mergeeESMap.put(JsonKey.USERNAME, null);
+            mergeeESMap.put(JsonKey.PREV_USED_EMAIL, mergee.getEmail());
+            mergeeESMap.put(JsonKey.PREV_USED_PHONE, mergee.getPhone());
+            mergeeESMap.put(JsonKey.UPDATED_DATE, ProjectUtil.getFormattedDate());
+            mergeeESMap.put(JsonKey.USER_ID, mergeeId);
```

Comment:
both mergeeDBMap and mergeeESMap have same set of data so just create one and use map.putAll

Reply:
yes code is redundant here, removed mergeeESMap completely and used the above one for cassandra and ES calls